docs/timezone.md

```diff
@@ -133,12 +135,67 @@ tz2 = Temporal.TimeZone.from(tz);
 ### timeZone.**id** : string
 
 The `id` property gives an unambiguous identifier for the time zone.
-Effectively, this is the canonicalized version of whatever `timeZoneIdentifier` was passed as a parameter to the constructor.
+This is the normalized version of whatever `timeZoneIdentifier` was passed as a parameter to the constructor.
 
 When subclassing `Temporal.TimeZone`, this property must be overridden to provide an identifier for the custom time zone.
 
 ## Methods
 
+### timeZone.**equals**(_other_: Temporal.TimeZone | object | string) : boolean
+
+**Parameters:**
+
+- `other` (`Temporal.TimeZone` object, object implementing the `Temporal.TimeZone` protocol, or a string time zone identifier): Another time zone to compare.
+
+**Returns:** `true` if `timeZone` and `other` are equal, or `false` if not.
+
+Compares two time zones for equality.
+Equality is determined by the following algorithm:
+
+- If `timeZone === other`, then the time zones are equal.
+- Otherwise, `timeZone.id` is compared to `other` (or `other.id` if `other` is an object).
+  If both string identifiers are equal, then the time zones are equal.
+  This is a case-sensitive comparison; custom time zone identifiers that vary in case will be considered unequal.
+- The time zones are also equal if the identifiers resolve to the same Zone name in the [IANA Time Zone Database](https://www.iana.org/time-zones).
+  This resolution is case-insensitive.
```

Comment:
Ugh, did we _really_ end up in a place where A and B are true but C is false?
```js
// equivalent strings
A: Temporal.TimeZone.from("Asia/Kolkata").equals("Asia/Calcutta");
// object with incorrect behavior that claims a tzdb id
B: Temporal.TimeZone.from("Asia/Kolkata").equals({
     id: "Fake Asia/Kolkata".split(" ").at(-1),
     getOffsetNanosecondsFor() { return 86400e9 - 1; },
     getPossibleInstantsFor() {},
   });
// proxy for a built-in time zone for a tzdb alias
C: Temporal.TimeZone.from("Asia/Kolkata").equals(new Proxy(
     Temporal.TimeZone.from("Asia/Calcutta"),
     { get(tz, k) { const v = tz[k]; return typeof v === "function" ? v.bind(tz) : v; } },
   ));
```

Reply:
Nope, they're all true. If both IDs are valid (case-insensitive) IDs in the IANA TZDB, then if they resolve to the same primary time zone identifier then they're considered equal. 

The case that won't work is when custom, non-IANA time zone IDs don't have the same case. Those are not considered equal. Like this:

```js
// Custom time zone identifiers are compared case-sensitively
class Custom1 extends Temporal.TimeZone {
  constructor() {  super('UTC'); }
  get id() { return 'Moon/Cheese'; }
};
class Custom2 extends Temporal.TimeZone {
  constructor() {  super('UTC'); }
  get id() { return 'Moon/CHEESE'; }
};
new Custom1().equals(new Custom1()); // => true
new Custom1().equals(new Custom2()); // => false
```

I agree that this is not ideal behavior, but when I discovered this the night before our Stage 3 presentations last week, @ptomato convinced me that it wasn't a big enough problem to warrant cracking open the reviewed-and-approved spec text.

Reply:
I reorganized the docs content to be more clear about this behavior, and I re-ordered the steps so IANA is first (because it's more common and more permissive). Let me know if you think the new docs text is better.

Reply:
> Nope, they're all true. If both IDs are valid (case-insensitive) IDs in the IANA TZDB, then if they resolve to the same primary time zone identifier then they're considered equal.

Got it. So there's a consistency, albeit of dubious value.

> The case that won't work is when custom, non-IANA time zone IDs don't have the same case. Those are not considered equal. Like this:
> 
> ```js
> // Custom time zone identifiers are compared case-sensitively
> class Custom1 extends Temporal.TimeZone {
>   constructor() {  super('UTC'); }
>   get id() { return 'Moon/Cheese'; }
> };
> class Custom2 extends Temporal.TimeZone {
>   constructor() {  super('UTC'); }
>   get id() { return 'Moon/CHEESE'; }
> };
> new Custom1().equals(new Custom1()); // => true
> new Custom1().equals(new Custom2()); // => false
> ```
> 
> I agree that this is not ideal behavior, but when I discovered this the night before our Stage 3 presentations last week, @ptomato convinced me that it wasn't a big enough problem to warrant cracking open the reviewed-and-approved spec text.

:+1:

> I reorganized the docs content to be more clear about this behavior, and I re-ordered the steps so IANA is first (because it's more common and more permissive). Let me know if you think the new docs text is better.

Much better; thanks.

Reply:
OK, cool. I also just added tests to cover case-sensitivity for custom objects. Resolving.